src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx

```diff
@@ -101,9 +104,54 @@ const RuleEditor = createReactClass({
     Promise.all(promises).then(([environments, rule]) => {
       this.setState({environments, rule});
     });
-  },
+  }
+
+  fetchStatus() {
+    this.pollHandler();
+    this.startPolling();
+  }
+
+  pollHandler = async () => {
+    const {api, organization, project} = this.props;
+    const {uuid} = this.state;
+    const origRule = {...this.state.rule};
+
+    const {status, rule} = await api.requestPromise(
+      `/projects/${organization.slug}/${project.slug}/rule-async-tasks/${uuid}/`
+    );
+
+    if (status === 'pending') {
+      return;
+    }
+
+    this.stopPolling();
+    if (status === 'failed') {
+      // TODO(meredith): better error message - maybe get the error
+      // passed through from the endpoint?
+      addErrorMessage(t('An error occurred'));
+      this.setState({loading: false, rule: origRule});
+    }
+    if (rule && status === 'success') {
+      this.setState({error: null, loading: false, rule});
+      const isNew = !origRule.id;
+      if (isNew) {
+        browserHistory.replace(
+          recreateRoute(`${rule.id}/`, {...this.props, stepBack: -1})
+        );
+      }
+      addSuccessMessage(isNew ? t('Created alert rule') : t('Updated alert rule'));
+    }
+  };
+
+  startPolling() {
+    this.intervalId = setInterval(this.pollHandler, 1000);
```

Comment:
@MeredithAnya Is this going to poll every second until the backend responds? We need to have a timeout on this polling. Maybe something like 3 minutes? I'm not sure what makes sense here.

Reply:
Be careful with setInterval() it makes no promises that you will get the gap you want. If API requests take more than one second to respond you could end up with two requests running at the same time.

A less racy approach is to use setTimeout() and queue the 'interval' at the end of each cycle. This generally let's you not need to worry about having to clear the interval later as well.

Reply:
+1 to using setTimout which can be coupled with a timer to make sure this doesn't go on forever.